### ISSUE TITLE
DirectX 11 support.

### ISSUE THREAD

**Issue by madurani** [cryinkfly/Autodesk-Fusion-360-for-Linux#504]
After installation of last version of Fusion appear there message, that DirectX 9 and OpenGL will be not supported any more in short time. I tried use DirectX 11, but without success. Windows stay black. Is there some possibility use DirectX 11 in wine(OpenGL work fine for me)?
![image](https://github.com/user-attachments/assets/9c271728-d740-431c-b0e4-edb164e2ae98)
![image](https://github.com/user-attachments/assets/3f758d69-9aa6-4db2-8172-425067dfa609)

**Comment by andrewljoy** [comment]
Same issue for me. Cannot get to the settings to even try DX11.

**Comment by madurani** [comment]
Summary of My Observations
OpenGL: Works fine, even without DXVK.
DirectX 9: Works fine without DXVK but stops rendering when DXVK is installed.
DirectX 11: The main window is black, whether or not DXVK is installed.

Has anyone configured DirectX11 with Fusion360 and rendering of main windows works properly?

I think that if vendor disable support for OpenGL and DirectX9, then we will all have problem.

**Comment by madurani** [comment]
Succeeded, I am on DirectX11. After replacing the vga from nvidia gtx660 to gtx1060 (support of vulkan 1.3 is requested). DXVK started working.

![image](https://github.com/user-attachments/assets/341352c9-a764-44e9-933f-e9414ef2e19f)

**Comment by madurani** [comment]
Nothing special: 
1.your vga hardware have to support vulkan1.3 (i have to replaced my old vga card)
2.install dxvk via winetricks
3.and after first login to fusion i changed graphical driver from opengl to directx11

A black screen flickering appeared. It should be fixed via dxvk.conf, but look like that wine ignore this conf file.

```
madurani@pc-marek1 ~/.PlayOnLinux/wineprefix/Fusion360_008/drive_c $ cat /volume01/wine/dxvk.conf
# Enable DirectX 11 support
dxgi.allow_unsupported = true
dxvk.enable_dxgi = true
dxvk.enable_d3d11 = true
dxvk.num_async_commands = 16
dxvk.enable_async = false
dxvk.use_raw_ssbo = false
dxvk.num_back_buffers = 3
dxvk.present_interval = 1
dxvk.shrink_cmdlists = true
dxvk.enable_rt = true
dxvk.use_vk_surface_extensions = true
dxvk.debug = info

```

```
madurani@pc-marek1 ~/.PlayOnLinux/wineprefix/Fusion360_008/drive_c $ vulkaninfo | grep -i version
Vulkan Instance Version: 1.3.296
VK_LAYER_KHRONOS_validation (Khronos Validation Layer) Vulkan version 1.3.296, layer version 1:
VK_LAYER_NV_optimus (NVIDIA Optimus layer) Vulkan version 1.3.277, layer version 1:
        apiVersion        = 1.3.277 (4206869)
        driverVersion     = 550.135.0.0 (2309079040)
        shaderBinaryVersion  = 1
        conformanceVersion:
        VK_KHR_sampler_ycbcr_conversion               : extension revision 14
        samplerYcbcrConversion             = true

```

**Comment by Holzniggl** [comment]
**Probably CUDA and ROCM related** 
**Nobara and Fedora 40**
**TLDR: try installing ROCM-Driver on AMD**
I got it running via DX11 on my new AMD-Tablett. Result was kinda sluggish. Black Screen flickering and such. not very kind to the eye. 
DXVK was automaticly installed via the Installscript and loaded. 
![Bildschirmfoto_20241210_202743](https://github.com/user-attachments/assets/09a7d4d7-4069-42c3-bc41-c02fc960c88e)
![Bildschirmfoto_20241210_220610](https://github.com/user-attachments/assets/8a698287-12c6-4ae8-82cd-090b9c3bd85e)
Since i'm using Nobara, i used their Driver Manager just to see how much i can mess around. 
Turns out there are Drivers for my GPU available. so i Installed it.
![Bildschirmfoto_20241210_221748](https://github.com/user-attachments/assets/a84d2d53-ea92-4f89-a348-5fd69168612c)
After a Reboot i can Confirm that i'm running on DX11 and  just the Browser and Comment tap are bugging out. When you detach those Windows and customize them in floating Mode, the black Screen goes away.
![Bildschirmfoto_20241210_221026](https://github.com/user-attachments/assets/7cf102fe-4f36-4252-a666-d6027c7d966c)

**Comment by techdregs** [comment]
I've got the same issue. DX11 just shows the editor window as black. Menus seem to work. I'm on F41 with Nvidia 565.57.01 drivers on Wayland. DXVK is installed via Winetricks. I haven't found any configuration that works. DX9 works fine.

`Dec 11 15:55:15 fedora env[39776]: 0e80:fixme:vulkan:X11DRV_vulkan_surface_create Application requires child window rendering, which is not implemented yet!
Dec 11 15:55:15 fedora env[39776]: err:   D3D11SwapChain: Failed to recreate surface: VK_ERROR_INCOMPATIBLE_DRIVER
Dec 11 15:55:15 fedora env[39776]: err:   DXGI: CreateSwapChainForHwnd: Failed to create swap chain, hr -2147467259
Dec 11 15:55:16 fedora env[39776]: warn:  DXGI: MakeWindowAssociation: Ignoring flags
`

**Comment by TheRob2D** [comment]
Hey wait, how are you guys changing to DX11????

When I click on preferences the process freezes entirely. Terminal shows: "QLayout: Attempting to add QLayout "" to QWidget "", which already has a layout" And I have to force close it.

**Comment by madurani** [comment]
yes, a preference windows start long, but when i minimalize/maximalize main windows of fusion, then a window immediate appear, or you can changed it in install script (this file is three times downloaded/copied to fusion folder, there is needed to change VirtualDeviceGLCore or VirtualDeviceDx9 to  VirtualDeviceDx11):

`wget -N https://raw.githubusercontent.com/cryinkfly/Autodesk-Fusion-360-for-Linux/main/files/setup/resource/video_driver/opengl/NMachineSpecificOptions.xml`

**Comment by TheRob2D** [comment]
Ah yes, I got it. Unfortunately, everything is too buggy to use. I'll stick to my VM for now. Great to see this project actually works though.

**Comment by madurani** [comment]
I've found that "black screen" appears when some popup menu is called or cursor is changed (when middle or right button of mouse is pressed), then it's enough just to scroll the mouse wheel and screen rendering is back.

After communication with chatgpt I got dxvk.conf which does not solver flickering on 100%, but now black screen show only for very short time and disappears by itself. In my case it is much better. 
Maybe someone who understands options in conf file can prepare some better configuration. 
```
cat /volume01/wine/dxvk.conf
# Enable DirectX 11 support
dxgi.allow_unsupported = true
dxvk.enable_dxgi = true
dxvk.enable_d3d11 = true
dxvk.num_async_commands = 16
dxvk.enable_async = false
dxvk.use_raw_ssbo = false
dxvk.num_back_buffers = 3
dxvk.present_interval = 1
dxvk.shrink_cmdlists = true
dxvk.enable_rt = true
dxvk.use_vk_surface_extensions = true
dxvk.debug = info
dxgi.maxFrameLatency = 1
dxgi.numBackBuffers = 2
d3d11.relaxedBarriers = True
dxgi.syncInterval = 1
dxvk.enableAsync = True
```
Here are variables which i used for start of fusion:

```
export WINEDEBUG="+vulkan,+dxgi,+d3d11"
export FUSION_IDSDK=false
export DXVK_FILTER_DEVICE_NAME="NVIDIA GeForce GTX 1060 3GB"
export DXVK_CONFIG_FILE=/volume01/wine/dxvk.conf
export WINEPREFIX="/home/madurani/.PlayOnLinux//wineprefix/Fusion360_010"
```

**Comment by onlyymax** [comment]
I couldn't solve it, do you have a 100% working solution?

**Comment by onlyymax** [comment]
I have the rtx 3070, and I have problems with directx 11 fusion, it is black and flickers

**Comment by ghost** [comment]
I accidentally found a decent improvement on this, like before you should have your driver set to DX11 and QT rendering on OpenGL. But for some reason after changing chromium graphics back-end to Vulkan, the screen dosent flicker nearly as much when an object is rendered. seems to only happen when you click off or use middle mouse button/scroll. Currently on fedora 42 KDE with AMD 7800xt. Not sure if its related to this however but Fusion 360 does not seem to close correctly after making this change leading to some weird interactions with my task bar. Ending task for CrBrowser main fixed this.

**Comment by Zeroji** [comment]
I managed to completely (so far 🤞) get rid of the flickering by hiding the navigation bar:

<img width="288" height="51" alt="Image" src="https://github.com/user-attachments/assets/b55653e3-cc96-4fef-adae-c56edef0400f" />

The shortcut to toggle its visibility is `Ctrl + Alt + N` by default, or you can find it in File > View.

**Comment by LYXOfficial** [comment]
> I managed to completely (so far 🤞) get rid of the flickering by hiding the navigation bar:我设法通过隐藏导航栏完全（到目前为止🤞）摆脱闪烁：
> 
> <img alt="Image" width="288" height="51" src="https://private-user-images.githubusercontent.com/15339858/483876565-b55653e3-cc96-4fef-adae-c56edef0400f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc4NDQyOTIsIm5iZiI6MTc1Nzg0Mzk5MiwicGF0aCI6Ii8xNTMzOTg1OC80ODM4NzY1NjUtYjU1NjUzZTMtY2M5Ni00ZmVmLWFkYWUtYzU2ZWRlZjA0MDBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE0VDA5NTk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3Yjk3NmUxMDlhNzQ4YzRiNmRiMGMzZmNjYWM4ZTMwZDAxOGQ4Zjc2NDlkYjM4YmYzZDRjZGM1OTM2ZDU3N2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CjCq0aPEzC4gzioRocPZYysieOAWhINeJ59wlaECm7U">
> The shortcut to toggle its visibility is `Ctrl + Alt + N` by default, or you can find it in File > View.默认情况下，切换其可见性的快捷键是 `Ctrl + Alt + N`，或者您可以在文件 > 视图中找到它。

It is a good idea to solve it.
When i am using dx9 or opengl the fusion starting logo covers the editor, so I have to use dx11, but it black and flickers seems so bad.
